### PR TITLE
[REVIEW] Add groupby size() method and fix groupby getattr with `as_index`

### PR DESCRIPTION
**Pull request by shwina**
cc: @randerzander 

1. Adds a `.size()` method to count the group sizes
2. Fixes groupby getattr/getitem to respect `as_index` argument

### CODE REVIEW

**Comment by jrhemstad** [comment]
How is `size` different than `count`?

**Comment by shwina** [comment]
It's just a convenience method to get the size of each group. 

```
In [3]: pdf                                                                                                                                                                                   
Out[3]: 
   a       b  c
0  1     bob  1
1  1     bob  2
2  3   alice  3
3  4  cooper  4

In [4]: pdf.groupby(['a', 'b']).size()                                                                                                                                                        
Out[4]: 
a  b     
1  bob       2
3  alice     1
4  cooper    1
dtype: int64

```

**Comment by jrhemstad** [comment]
> It's just a convenience method to get the size of each group.

Looks like you're materializing a column of 64bit `1`s and then computing a `sum` agg. It'd be more efficient to instead do a temp column of 8 bit values and do a `count` agg. 

I'm realizing for something like this that it'd be nice if you could call `cudf::groupby` with just keys and have the values be empty.

File: python/cudf/cudf/groupby/groupby.py [hunk 1]
```diff
@@ -44,10 +45,19 @@ def count(self):
     def agg(self, func):
         return self._apply_aggregation(func)
 
+    def size(self):
+        nrows = len(self._groupby.obj)
+        ones = cudf.Series(cudautils.ones(nrows, dtype="int8"))
```
Comment (shwina) [inline python/cudf/cudf/groupby/groupby.py]: Can this just be an uninitialized column for efficiency?

Comment (jrhemstad) [reply python/cudf/cudf/groupby/groupby.py]: yeah, it can

**Pull request merged**